keps/sig-api-machinery/2885-server-side-unknown-field-validation/README.md

```diff
@@ -696,20 +692,22 @@ enhancement:
   cluster required to make on upgrade, in order to maintain previous behavior?
 - What changes (in invocations, configurations, API use, etc.) is an existing
   cluster required to make on upgrade, in order to make use of the enhancement?
+  -->
 
 ### Version Skew Strategy
 
-If applicable, how will the component handle version skew with other
-components? What are the guarantees? Make sure this is in the test plan.
+Following the graduation of server-side field validation to GA, there will be a
+period of time when a newer client (which expects server-side field validation locked in GA) will send
+requests to an older server that could have server-side field validation
+disabled. Until version skew makes it incompatible for a client to hit a server
+with field validation disabled, kubectl will need to continue to have
+client-side validation available. 
 
-Consider the following in developing a version skew strategy for this
-enhancement:
-- Does this enhancement involve coordinating behavior in the control plane and
-  in the kubelet? How does an n-2 kubelet without this feature available behave
-  when this feature is used?
-- Will any other components on the node change? For example, changes to CSI,
-  CRI or CNI may require updating that component before the kubelet.
--->
+After this skew (2-3 releases), client-side
```

Comment:
Why 2-3? Officially, kubectl is only supposed to support 1 version skew.

Reply:
if the cost is low, I'd be open to keeping the client-side validation in place until 1.26 goes out of support, even if it's not strictly required

Reply:
though... server-side validation has been on by default since 1.25, and I can't find evidence anyone has disabled it, so dropping client-side validation in 1.28 would be 1-version-skew-compatible with GA in 1.27 and practically would mean all OSS-supported versions as of August 2023 (1.25+) would have server-side validation enabled by default.

Reply:
> August ~2023~2022

Reply:
Yeah, as I mentioned below, there are two things: 
1. do we stop checking if server-side strict validation is enabled on the server (I'd like that to stop as early as possible because it forces a pull on the openapi every single time).
2. do we stop support client-side validation at all? That is probably up to sig-cli to decide whether they are happy with the alternatives. That still requires a pull on openapi v2 but would only happen when someone specifically asks for it (which should be much more occasional).

Reply:
>> all OSS-supported versions as of August 2023 (1.25+)
>
> August ~2023~2022

August 2023~2022~ :)

1.28 is estimated to release in August 2023, at which point 1.25+ will be still in support :)